docs/02-architecture/06-client-side-architecture.md

```diff
@@ -0,0 +1,285 @@
+## Javascript
+
+### Quick Overview
+
+The JS app can be split into distinct parts:
+
+- Inline blocking JS
+- Inline Non-blocking JS
+- app.js - curl module loader, the bundle booter and the standard Javascript
+- Commercial Javascript
+- Enhanced Javascript
+
+And then specific Javascript bootstraps used on different pages and required into the app when needed - Article, Article Minute, Crosswords, Liveblog, Gallery, Trail, Profile, Sudoku, Image content, Facia, Football, Preferences, Membership, Ophan, Admin, Main Media, Video Embed, Accessibility.
+
+### General structure
+
+See below for quick descriptions.
+
+- [- main.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/main.scala.html)
+	- [head.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/head.scala.html)
+		- [inlineJsBlocking.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/inlineJSBlocking.scala.html) - Scripts required to render the page correctly on first paint.
+			- Polyfills for json2, es5-html5, RAF, classlist
+			- Curl config
+			- shouldEnhance
+			- pageConfig
+			- applyRenderConditions
+			- loadFonts
+			- enable non blocking stylesheets
+			- **Load the main app async**
+				- app.js
+				- commercial.js
+				- enhanced/main.js
+			- Cloudwatch beacon
+		- [inlineJSNonBlocking.scala.html](https://github.com/guardian/frontend/blob/master/common/app/views/fragments/inlineJSNonBlocking.scala.html)
+			- getUserData.js
+			- detectAdblock 
+			- showUserName
+			- ophanConfig
+		
```

Comment:
Should we also mention [analytics loading](https://github.com/guardian/frontend/blob/master/common/app/views/main.scala.html#L112) here?